### PR TITLE
refactor: minor fixes on kubeadm deployment scripts

### PR DESCRIPTION
**Pull request by alouvion-aneo**
fix some feedbacks about previous merge : https://github.com/aneoconsulting/ArmoniK.Infra/pull/4
add some output logs
rename some variables

### CODE REVIEW

File: kubernetes/onpremise/kubeadm/README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ No modules.
 | <a name="input_master_node_name"></a> [master\_node\_name](#input\_master\_node\_name) | The name of the cluster master node. | `string` | `"master"` | no |
 | <a name="input_master_private_ip"></a> [master\_private\_ip](#input\_master\_private\_ip) | The private ip of the master node. | `string` | n/a | yes |
 | <a name="input_master_public_ip"></a> [master\_public\_ip](#input\_master\_public\_ip) | The public ip of the master node. | `string` | n/a | yes |
-| <a name="input_tls_private_key_pem"></a> [tls\_private\_key\_pem](#input\_tls\_private\_key\_pem) | The private key of the master node. | `string` | n/a | yes |
+| <a name="input_tls_private_key_pem_content"></a> [tls\_private\_key\_pem\_content](#input\_tls\_private\_key\_pem\_content) | The private key of the master node. | `string` | n/a | yes |
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/README.md:48]: Maybe you should precise in the description that it is the content of the private key and not its name ?

Comment (alouvion-aneo) [reply kubernetes/onpremise/kubeadm/README.md:48]: yes

File: kubernetes/onpremise/kubeadm/examples/complete/README.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 | Name | Version |
 |------|---------|
+| <a name="provider_null"></a> [null](#provider\_null) | n/a |
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/README.md:13]: Add minimal version

File: kubernetes/onpremise/kubeadm/examples/complete/main.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    },
+    "worker-2" = {
+      "name"                = "worker-2"
+      "labels"              = tolist(["workers", ])
+      "taints"              = tolist(["workers", ])
+      "public_dns"          = "PUBLIC_DNS_HERE" # it can be private if you are in the dest network
+      "tls_private_key_pem" = "my worker-2 node ssh private key"
+    }
+  }
+  user  = "ec2-user" #default redhat user
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/main.tf:25]: Put these variables, which have fake values, in varaibles.tf and parameters.auto.tfvars files. Explain in the README.md that the user must update the values of the configuration in the parameters file.

File: kubernetes/onpremise/kubeadm/examples/complete/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+output "kubeconfig_file" {
+  description = "Use this export to begin to use your cluster"
+  value       = "export KUBECONFIG=$HOME/.kube/config_remote"
+}
+
+output "kubeconfig_file" {
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/outputs.tf:6]: Both output values have the same name ?

File: kubernetes/onpremise/kubeadm/examples/complete/README.md (hunk 2 above)
Comment (alouvion-aneo) [reply kubernetes/onpremise/kubeadm/examples/complete/README.md:13]: ok

File: kubernetes/onpremise/kubeadm/examples/complete/outputs.tf (hunk 4 above)
Comment (alouvion-aneo) [reply kubernetes/onpremise/kubeadm/examples/complete/outputs.tf:6]: ok

File: kubernetes/onpremise/kubeadm/examples/complete/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+| Name | Version |
+|------|---------|
+| <a name="provider_null"></a> [null](#provider\_null) | >= 3.2.1 |
+| <a name="provider_random"></a> [random](#provider\_random) | ~> 3.5.1 |
+
+## Modules
+
+| Name | Source | Version |
+|------|--------|---------|
+| <a name="module_install_kubeadm_cluster"></a> [install\_kubeadm\_cluster](#module\_install\_kubeadm\_cluster) | ../.. | n/a |
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/README.md:21]: add the name of the module

File: kubernetes/onpremise/kubeadm/examples/complete/README.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# OnPremise kubeadm
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/README.md:1]:
```suggestion
# Onpremise Kubeadm
```

File: kubernetes/onpremise/kubeadm/examples/complete/README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+### how to use it : 
+You have to update values PUBLIC_DNS_HERE, PRIVATE_DNS_HERE and TLS_PRIVATE_KEY_FILE_HERE
+in the file [terraform.tfvars](terraform.tfvars)
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/README.md:4]:
```suggestion
The Terraform scripts use the module [kubeadm](../../kubeadm) to install and configure a Kubernetes cluster with `Kubeadm`.

First, you should have a list of barre metal ou virtual machines, and then you have to update values `PUBLIC_DNS_HERE`, `PRIVATE_DNS_HERE` and `TLS_PRIVATE_KEY_FILE_HERE` in the file [terraform.tfvars](terraform.tfvars).
```

File: kubernetes/onpremise/kubeadm/examples/complete/main.tf [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+# call the module for kubeadm cluster creation
+module "install_kubeadm_cluster" {
+  source = "../../../kubeadm"
+
+  master_public_ip            = var.master.public_dns
+  master_private_ip           = var.master.private_dns
+  tls_private_key_pem_content = file(var.master.tls_private_key_pem_file)
+  user                        = var.user
+  kubeadm_token               = local.token
+  cni_pluggin                 = "flannel" # calico or flannel
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/main.tf:15]: Did you use Terraform validation for variable `cni_pluggin` to declare the valid values `calico` and `flannel`?

File: kubernetes/onpremise/kubeadm/examples/complete/main.tf [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  loadbalancer_plugin         = "metalLB"
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/main.tf:16]: Did you use Terraform validation for variable `loadbalancer_plugin` to declare the valid values `metalLB` ?

File: kubernetes/onpremise/kubeadm/examples/complete/README.md (hunk 5 above)
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/README.md]:
```suggestion
| <a name="module_install_kubeadm_cluster"></a> [install\_kubeadm\_cluster](#module\_install\_kubeadm\_cluster) | ../../kubeadm | n/a |
```

File: kubernetes/onpremise/kubeadm/examples/complete/main.tf [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+      "tls_private_key_pem" = "my worker-2 node ssh private key"
+    }
+  }
+  user  = "ec2-user" #default redhat user
+  token = "${random_string.token_id.result}.${random_string.token_secret.result}"
+}
+
+# call the module for kubeadm cluster creation
+module "install_kubeadm_cluster" {
+  source                      = "../.."
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/examples/complete/main.tf]:
```suggestion
  source                      = "../../kubeadm"
```

**Pull request merged**